client/src/plugins/camunda-plugin/deployment-tool/DeploymentTool.js

```diff
@@ -129,26 +131,32 @@ export default class DeploymentTool extends PureComponent {
     }
 
     // (3) Trigger deployment
-    // (3.1) Show deployment result (success or error)
+    let version;
 
     try {
+
+      // (3.1) Retrieve version we deploy to via API
+      version = (await this.getVersion(configuration)).version;
```

Comment:
Can we ensure the response won't be undefined?

I can see a test case that indicated that it might be possible: https://github.com/camunda/camunda-modeler/pull/2228/commits/0cc258bb9f45d9cb538097364f48b328c036767b#diff-d09ace10bd1c982913d688081febd03b45ea9de6856b302d5396a2ff246545b9R403

Reply:
if it can be undefined, what shall we do? Failing and end the deployment or at least try to deploy?

Reply:
Yep, so this is a very theoretical edge case I believe.

Generally:

* If `#getVersion` does not throw, it returns a valid version
* Theoretically, if the engine does not provide a `version`-value in a `200` response, it would return undefined. We then have two options:
  * (A) Throw
  * (B) Return undefined 
* I decided for (B) since the engine is still in a valid `200` state and I would hence not like to "block" the deployment by manually throwing. In this case, a version `undefined` would be send to `E.T.`, which is okay I believe.

BTW: 
> if the engine does not provide a `version`-value in a `200` response

This would be a bug in the engine, so we could also remove this test case: https://github.com/camunda/camunda-modeler/commit/0cc258bb9f45d9cb538097364f48b328c036767b#diff-d09ace10bd1c982913d688081febd03b45ea9de6856b302d5396a2ff246545b9R403

WDYT?

Reply:
> I decided for (B) since the engine is still in a valid 200 state and I would hence not like to "block" the deployment by manually throwing. 

I agree 👍  I just wanted to make sure that we don't fail because of this line. As `CamundaAPI#getVersion` always returns an object in terms of a 200 response, it should be fine, also in case, the engine does not return anything in a successful state (which is not something we should bother on our side).

```js
{
  version: undefined
}
```